ui/components/ui/box/box.js

```diff
@@ -104,54 +88,113 @@ export const ValidBorderColors = [
+  return typeof type === 'string' && typeof value === 'string';
+}
 
-  const isHorizontalAndVertical = arr.length === 2;
-  const isTopHorizontalAndBottom = arr.length === 3;
-  const isAllFour = arr.length === 4;
-  const hasAtLeastTwo = arr.length >= 2;
-  const hasAtLeastThree = arr.length >= 3;
+/**
+ * Generates classnames for different utility styles from utility props such as margin, padding, display, etc.
+ * Accepts numbers and strings for static values and arrays for responsive values.
+ * Maps responsive props to mobile first breakpoints
+ * Returns an object for 'classnames' package to generate the utility classNames
+ *
+ * @param {string} baseClass - The root or base class name
+ * @param {string} type - The style declaration type "margin", "margin-top", "padding", "display" etc
+ * @param {array || number || string} value - prop value being passed in array props are responsive props
+ * @param {*} validatorFn - The validation function for each value type
+ * @returns
+ */
 
-  return {
-    [`${baseClass}--${type}-${singleDigit}`]: isValidValue(type, singleDigit),
-    [`${baseClass}--${type}-top-${top}`]: isValidValue(type, top),
-    [`${baseClass}--${type}-right-${right}`]: isValidValue(type, right),
-    [`${baseClass}--${type}-bottom-${bottom}`]: isValidValue(type, bottom),
-    [`${baseClass}--${type}-left-${left}`]: isValidValue(type, left),
-    // As long as an array of length >= 2 has been provided, the first number
-    // will always be for the top value.
-    [`${baseClass}--${type}-top-${arr?.[0]}`]: hasAtLeastTwo,
-    // As long as an array of length >= 2 has been provided, the second number
-    // will always be for the right value.
-    [`${baseClass}--${type}-right-${arr?.[1]}`]: hasAtLeastTwo,
-    // If an array has 2 values, the first number is the bottom value. If
-    // instead if has 3 or more values, the third number will be the bottom.
-    [`${baseClass}--${type}-bottom-${arr?.[2]}`]: hasAtLeastThree,
-    [`${baseClass}--${type}-bottom-${arr?.[0]}`]: isHorizontalAndVertical,
-    // If an array has 2 or 3 values, the second number will be the left value
-    [`${baseClass}--${type}-left-${arr?.[1]}`]:
-      isHorizontalAndVertical || isTopHorizontalAndBottom,
-    // If an array has 4 values, the fourth number is the left value
-    [`${baseClass}--${type}-left-${arr?.[3]}`]: isAllFour,
-  };
+function generateClassNames(baseClass, type, value, validatorFn) {
+  // if value does not exist return null
+  if (!value) {
+    return null;
+  }
+  // initiate classObject to be returned to classnames package
+  let classesObject = {};
+  // check if prop is not an array prop
+  let singleDigit = Array.isArray(value) ? undefined : value;
+  // check if single digit exists or if array has only one item
+  if (singleDigit || value.length === 1) {
+    // if it is an array with only one item assign it to singleDigit
+    if (value.length === 1) {
+      singleDigit = value[0];
+    }
```

Comment:
I tried both of the above but because `.length` will work on a string. It would return the first letter of any string value. Attempted to simplify here https://github.com/MetaMask/metamask-extension/pull/15106/commits/2f49d61f8b00634b6dd94b1e919acef118b0775c